internal/merger/merger_test.go

```diff
@@ -78,6 +79,155 @@ func TestMergeResolve(t *testing.T) {
 	}
 }
 
+func TestMergeNode(t *testing.T) {
+	cmpOpts := cmp.Options{
+		protocmp.Transform(),
+	}
+
+	for _, c := range []struct {
+		local                *pbv2.NodeResponse
+		remote               *pbv2.NodeResponse
+		want                 *pbv2.NodeResponse
+		localPaginationInfo  *pbv1.PaginationInfo
+		remotePaginationInfo *pbv1.PaginationInfo
+		wantPaginationInfo   *pbv1.PaginationInfo
+	}{
+		{
+			&pbv2.NodeResponse{
+				Data: map[string]*pbv2.LinkedGraph{
+					"dcid1": {
+						Arcs: map[string]*pbv2.Nodes{
+							"prop1": {
+								Nodes: []*pb.EntityInfo{
+									{Value: "val1"},
+								},
+							},
+						},
+					},
+				},
+			},
+			&pbv2.NodeResponse{
+				Data: map[string]*pbv2.LinkedGraph{
+					"dcid2": {
```

Comment:
maybe also test adding dcid1 in remote and combine nodes from local and remote?

Reply:
done